Library/Homebrew/utils/popen.rb

```diff
@@ -1,22 +1,47 @@
-# typed: true
+# typed: strict
 # frozen_string_literal: true
 
 module Utils
+  extend T::Sig
+
   IO_DEFAULT_BUFFER_SIZE = 4096
   private_constant :IO_DEFAULT_BUFFER_SIZE
 
+  sig {
+    params(
+      args:    T.any(Pathname, String, T::Hash[String, String]),
```

Comment:
The `T::Hash` is because the first argument to `popen`/`exec` can be a hash of ENV vars. I attempted to encode this (see the first commit) but decided it was too messy once I realized the downstream consequences.